CenterForOpenScience/osf.io
# [ENG-2110] Use custom DOI prefixes for registries providers

## Purpose

Currently certain communities have there own special DOI prefixes, and now registries moderation should have it too!

## Changes

- changes client to send custom prefix
- changes tests to work with less mockable client

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify the datacite test site is getting the custom prefix when set for a registration provider via the admin app.

## Documentation

tech task, no new docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2110
